core/gsb-api/src/lib.rs

```diff
@@ -0,0 +1,756 @@
+        let response = flexbuffer_util::as_map(&response, false)
+            .map_err(|err| format!("Missing root map. Err: {err}"))?;
+        let id = flexbuffer_util::read_string(&response, "id")
+            .map_err(|err| format!("Missing response id. Err: {err}"))?;
+        if let Ok(error_payload) = flexbuffer_util::read_map(&response, "error", false) {
+            WsResponse::try_new("Err", &id, error_payload)
+                .map_err(|err| format!("Failed to read error payload. Id: {id}. Err: {err}"))
+        } else if let Ok(payload) = flexbuffer_util::read_map(&response, "payload", true) {
+            WsResponse::try_new("Ok", &id, payload)
+                .map_err(|err| format!("Failed to read payload. Id: {id}. Err: {err}"))
+        } else {
+            Err(format!("Missing 'payload' and 'error' fields. Id: {id}."))
+        }
+    }
+
+    fn start_buffering(
+        &self,
+        reason: Option<CloseReason>,
+        ctx: &mut WebsocketContext<WsMessagesHandler>,
+    ) {
+        log::debug!("WS Close. Reason: {reason:?}");
+        let drop_messages = match reason {
+            Some(reason) => DropMessages { reason },
+            None => {
+                let code = CloseCode::Normal;
+                let description = Some("Closing".to_string());
+                let reason = CloseReason { code, description };
+                DropMessages { reason }
+            }
+        };
+        let drop_messages_fut = self.service.send(drop_messages).boxed();
+        let start_buffering_fut = self.service.send(StartBuffering).boxed();
+        ctx.wait(actix::fut::wrap_future(async {
+            if let Err(error) = drop_messages_fut.await {
+                log::error!("Failed to send DropMessages. Err: {}", error);
+            }
+            if let Err(error) = start_buffering_fut.await {
+                log::error!("Failed to send StartBuffering. Err: {}", error);
+            }
+        }));
+    }
+
+    fn close(&self, ctx: &mut WebsocketContext<WsMessagesHandler>, code: CloseCode, desc: &str) {
+        let description = Some(desc.to_string());
+        let reason = Some(CloseReason { code, description });
+        self.start_buffering(reason.clone(), ctx);
+        log::warn!("Closing WS handler: {}", desc);
+        ctx.close(reason);
+    }
+}
+
+impl Actor for WsMessagesHandler {
+    type Context = ws::WebsocketContext<Self>;
+
+    fn started(&mut self, _ctx: &mut Self::Context) {
+        log::debug!("WsMessagesHandler started");
+    }
+
+    fn stopped(&mut self, _ctx: &mut Self::Context) {
+        log::debug!("WsMessagesHandler stopped");
+    }
+}
+
+impl Handler<WsRequest> for WsMessagesHandler {
+    type Result = <WsRequest as Message>::Result;
+
+    fn handle(&mut self, request: WsRequest, ctx: &mut Self::Context) -> Self::Result {
+        log::info!(
+            "WS request (id: {}, component: {})",
+            request.id,
+            request.component
+        );
```

Comment:
I don't think you would like to see each request on info level

Reply:
Changed to debug.